docs/interactions/Application_Commands.md

```diff
@@ -1,83 +1,130 @@
+> Commands can only be registered via HTTP endpoint.
 
-There are two kinds of Slash Commands: global commands and guild commands. Global commands are available for every guild that adds your app. An individual app's global commands are also available in DMs if that app has a bot that shares a mutual guild with the user.
+Commands can be scoped either globally or to a specific guild. Global commands are available for every guild that adds your app. An individual app's global commands are also available in DMs if that app has a bot that shares a mutual guild with the user.
 
-Guild commands are specific to the guild you specify when making them. Guild commands are not available in DMs. Command names are unique per application within each scope (global and guild). That means:
+Guild commands are specific to the guild you specify when making them. Guild commands are not available in DMs. Command names are unique per application, per type, within each scope (global and guild). That means:
 
-- Your app **cannot** have two global commands with the same name
-- Your app **cannot** have two guild commands within the same name **on the same guild**
-- Your app **can** have a global and guild command with the same name
+- Your app **cannot** have two global `CHAT_INPUT` commands with the same name
+- Your app **cannot** have two guild `CHAT_INPUT` commands within the same name **on the same guild**
+- Your app **can** have a global and guild `CHAT_INPUT` command with the same name
+- Your app **can** have a global `CHAT_INPUT` and `USER` command with the same name
 - Multiple apps **can** have commands with the same names
 
-> info
-> Apps can have a maximum of 100 global commands, and an additional 100 guild-specific commands per guild
+An app can have the following number of commands:
 
-**Command names must be lower-case** and match the regular expression `^[\w-]{1,32}$`. Commands (including sub-commands) with upper- or mixed- case names will be rejected by the API with a HTTP 400 (Bad Request) response.
+- 100 global `CHAT_INPUT` commands
+- 5 global `USER` commands
+- 5 global `MESSAGE` commands
+  
+As well as the same amount of guild-specific commands per guild.
 
-To make a **global** Slash Command, make an HTTP POST call like this:
+> danger
+> There is a global rate limit of 200 application command creates per day, per guild
+
+### Making a Global Command
+
+Global commands are available on _all_ your app's guilds. Global commands are cached for **1 hour**. That means that new global commands will fan out slowly across all guilds, and will be guaranteed to be updated in an hour.
+
+Global commands have inherent read-repair functionality. That means that if you make an update to a global command, and a user tries to use that command before it has updated for them, Discord will do an internal version check and reject the command, and trigger a faster reload for the guild in which it was used.
```

Comment:
Any details on how much "faster" this "faster reload" is?

Reply:
Let me find out exactly.

Reply:
It'll get refreshed next time a client queries for commands, but there's a rate limit on it, so it's possible that a user could have a command rejected a couple times until a rate limit is lifted.

That's all behind the scenes semantics though. The TLDR is that if you update a command, you can be sure that you will receive interactions payloads that reflect the updated command.